src/Rope.elm

```diff
@@ -0,0 +1,51 @@
+module Rope exposing (Rope(..), empty, flatFromList, fromList, one, prependTo, toList)
+
+{-| inspired by [miniBill/elm-rope](https://dark.elm.dmy.fr/packages/miniBill/elm-rope/latest/)
+-}
+
+
+type Rope a
+    = Leaf (List a)
+    | Branch (List (Rope a))
+
+
+fromList : List a -> Rope a
+fromList list =
+    Leaf list
+
+
+empty : Rope a
+empty =
+    Leaf []
+
+
+one : a -> Rope a
+one onlyElement =
+    Leaf (List.singleton onlyElement)
```

Comment:
Unless the compiler inlines `List.singleton`, which I doubt, we should replace this by `Lead [ onlyElement ]`. It's simpler and faster.

Reply:
fixed, thanks.